internal/db/fetcher/versioned.go

```diff
@@ -182,7 +181,7 @@ func (vf *VersionedFetcher) Start(ctx context.Context, spans core.Spans) error {
 	}
 
 	vf.ctx = ctx
-	vf.dsKey = dk
+	vf.dsKey = dk.WithCollectionRoot(vf.col.Description().RootID)
```

Comment:
question: This change is unexpected and unexplained. Can you describe why this was needed?

Reply:
Ah sorry - I see this was fairly inexplicable from the PR - previously this was done (in quite a complex way) by the `base.MakePrimaryIndexKeyForCRDT` (within the iteration loop).  Pulling it out here aided the simplification of `processBlock`.